Data engineering basics

AlSovPich opened the pull request.

AlSovPich committed:
Uploading basic version of data_engineering

AlSovPich committed:
Update main.py

AlSovPich committed:
Update data_preparation.py

abrosimov-software commented on stages/data_engineering/src/data_extraction.py:
```diff
@@ -1 +1,16 @@
+def extract_data(source: Optional[str] = None):
+    current_dir = os.path.dirname(os.path.abspath(__file__))
+    if source == None:
+        audiofiles_dir = os.path.join(current_dir, '../../../data/raw/audiofiles')
```
Extracted files must be saved within data_engineering microservice. Here extract_data attempts to access folder from outside of its domain. 
In docker-compose.yaml at root of the project we provide the data_engineering container with /app/data/preprocessed volume, which links the projects folder.
For now, it would be better to change the data paths to /app/data/|something|

abrosimov-software commented on stages/data_engineering/src/data_preparation.py:
```diff
@@ -1 +1,18 @@
+    normalized_data = [] 
+
+    current_dir = os.path.dirname(os.path.abspath(__file__))
+    audiofiles_dir = os.path.join(current_dir, '../../../data/preprocessed/audiofiles')
```
Same here. Microservice does not have access to the "../../../data" directly.
We grant this access by mapping the projects "data/|something|" to microservice's "/app/data/|something|" in docker-compose.yaml

abrosimov-software commented on stages/data_engineering/main.py:
This is correct, but maybe it would be better to separate a bit more:
- One function extracts the data for the microservice and only returns status or exception (doesn't take any parameters)
- Another function takes string as parameter (e.g. "raw", "processed") and returns corresponding dataset

AlSovPich committed:
Add files via upload

AlSovPich committed:
Remade files

Changed data extraction and preparation format
Added feature and label extraction
Should be compatible with model_engineering code

AlSovPich committed:
Code rewritten, neutral-normalization implemented

Now to avoid actor voice affecting predictions, for each actor features for neutral expression are computed and used to normalize all the other actor recordings
